lib/common/xml.c

```diff
@@ -1206,45 +1207,46 @@ write_xml_stream(const xmlNode *xml, const char *filename, FILE *stream,
                  bool compress, unsigned int *nbytes)
 {
     // @COMPAT Drop nbytes as arg when we drop write_xml_fd()/write_xml_file()
-    int rc = pcmk_rc_ok;
-    unsigned int bytes_out = 0;
     char *buffer = NULL;
+    unsigned int bytes_out = 0;
+    int rc = pcmk_rc_ok;
 
     crm_log_xml_trace(xml, "writing");
 
     buffer = dump_xml_formatted(xml);
-    CRM_CHECK(buffer && strlen(buffer),
-              crm_log_xml_warn(xml, "formatting failed");
+    CRM_CHECK(!pcmk__str_empty(buffer),
+              crm_log_xml_info(xml, "dump-failed");
               rc = pcmk_rc_error;
               goto bail);
 
     if (compress) {
         unsigned int bytes_in = 0;
-        BZFILE *bz_file = NULL;
 
-        rc = BZ_OK;
-        bz_file = BZ2_bzWriteOpen(&rc, stream, 5, 0, 30);
-        rc = pcmk__bzlib2rc(rc);
+        // (5, 0, 0): (intermediate block size, silent, default workFactor)
+        BZFILE *bz_file = BZ2_bzWriteOpen(&rc, stream, 5, 0, 0);
```

Comment:
default is 30; I'll note that in the commit message. wanted to avoid magic numbers as much as possible